tests/tracing/test_dnacurvature.py

```diff
@@ -0,0 +1,941 @@
+                    [-1.06762746e00, 3.28581945e00],
+                    [1.06762746e00, 3.28581945e00],
+                    [2.79508497e00, 2.03074810e00],
+                ]
+            ),
+            np.asarray([0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]),
+        ),
+        (
+            10,
+            10,
+            2,
+            0,
+            np.asarray(
+                [
+                    [-4.44089210e-16, 2.93892626e00],
+                    [1.72745751e00, 2.37764129e00],
+                    [2.79508497e00, 9.08178160e-01],
+                    [2.79508497e00, -9.08178160e-01],
+                    [1.72745751e00, -2.37764129e00],
+                    [8.88178420e-16, -2.93892626e00],
+                    [-1.72745751e00, -2.37764129e00],
+                    [-2.79508497e00, -9.08178160e-01],
+                    [-2.79508497e00, 9.08178160e-01],
+                    [-1.72745751e00, 2.37764129e00],
+                ]
+            ),
+            np.asarray(
+                [
+                    [8.63728757e-01, -1.11022302e-16],
+                    [6.98771243e-01, -5.07687025e-01],
+                    [2.66906864e-01, -8.21454863e-01],
+                    [-2.66906864e-01, -8.21454863e-01],
+                    [-6.98771243e-01, -5.07687025e-01],
+                    [-8.63728757e-01, -1.11022302e-16],
+                    [-6.98771243e-01, 5.07687025e-01],
+                    [-2.66906864e-01, 8.21454863e-01],
+                    [2.66906864e-01, 8.21454863e-01],
+                    [6.98771243e-01, 5.07687025e-01],
+                ]
+            ),
+            np.asarray([0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]),
+        ),
+    ],
+)
+def test_curvature_circle(
+    points,
+    radius,
+    edge_order,
+    expected_variance,
+    expected_first_derivative,
+    expected_second_derivative,
+    expected_local_curvature,
+) -> None:
+    """Test first and second derivatives and local curvature of a circle.
+
+    The variance of local curvature is also tested, for a circle it should be zero."""
+    circle_points = circle_coordinates(points, radius)
+    curvature = Curvature(molecule_coordinates=circle_points, circular=True)
+    curvature.calculate_derivatives(edge_order=edge_order)
+    curvature._calculate_local_curvature()
+    np.testing.assert_array_almost_equal(curvature.first_derivative, expected_first_derivative)
+    np.testing.assert_array_almost_equal(curvature.second_derivative, expected_second_derivative)
+    np.testing.assert_array_almost_equal(curvature.local_curvature, expected_local_curvature)
+    np.testing.assert_almost_equal(np.var(curvature.local_curvature), expected_variance)
+
+
+@pytest.mark.parametrize(
+    "points, radius, shift, edge_order, expected_variance, expected_first_derivative, expected_second_derivative, expected_local_curvature",
+    [
+        (
+            10,
+            10,
+            5,  # Shift array by 3 points
```

Comment:
Why does this say 5 in the code but 3 in the comment?

Reply:
Thanks :+1: corrected.